File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/physics/shapes/ConvexTriangleMeshShapeJson.java
package io.github.movementspeed.nhglib.data.models.serialization.physics.shapes;

import com.badlogic.gdx.utils.JsonValue;
import io.github.movementspeed.nhglib.physics.models.ConvexTriangleMeshRigidBodyShape;

/**
 * Created by Fausto Napoli on 15/06/2017.
 */
public class ConvexTriangleMeshShapeJson extends ShapeJson {
    private ConvexTriangleMeshRigidBodyShape shape;

    @Override
    public void parse(JsonValue jsonValue) {
        boolean calcAabb = jsonValue.getBoolean("calcAabb", true);
        String asset = jsonValue.getString("asset", "");

        shape = new ConvexTriangleMeshRigidBodyShape(asset, calcAabb);
    }

    @Override
    public ConvexTriangleMeshRigidBodyShape get() {
        return shape;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/physics/shapes/ConvexTriangleMeshShapeJson.kt
package io.github.movementspeed.nhglib.data.models.serialization.physics.shapes

import com.badlogic.gdx.utils.JsonValue
import io.github.movementspeed.nhglib.physics.models.ConvexTriangleMeshRigidBodyShape

/**
 * Created by Fausto Napoli on 15/06/2017.
 */
class ConvexTriangleMeshShapeJson : ShapeJson() {
    private var shape: ConvexTriangleMeshRigidBodyShape? = null

    override fun parse(jsonValue: JsonValue) {
        val calcAabb = jsonValue.getBoolean("calcAabb", true)
        val asset = jsonValue.getString("asset", "")

        shape = ConvexTriangleMeshRigidBodyShape(asset, calcAabb)
    }

    override fun get(): ConvexTriangleMeshRigidBodyShape? {
        return shape
    }
}
